### PR TITLE
Miner actor submit window post v0

### PR DESCRIPTION
**Pull request by wer1st**
### Description of the Change

- Implement Submit Window PoSt of Miner Actor
- Fix RleBitset codec
- Fix Amt
- Stop mechanism for amt visitors

### CODE REVIEW

File: core/vm/actor/builtin/types/miner/sectors.cpp [hunk 1]
```diff
@@ -64,6 +64,15 @@ namespace fc::vm::actor::builtin::types::miner {
     return sector_infos;
   }
 
+  outcome::result<Sectors> Sectors::loadSectors() const {
+    const auto sectors_copy = *this;
+
+    // Lotus gas conformance
+    OUTCOME_TRY(sectors_copy.sectors.amt.loadRoot());
```
Comment (turuslan) [inline core/vm/actor/builtin/types/miner/sectors.cpp:71]: assert that amt is cid

File: core/adt/stop.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "common/outcome.hpp"
+
+#define CATCH_STOP(expr) OUTCOME_TRY(adt::catchStop((expr)))
```
Comment (turuslan) [inline core/adt/stop.hpp:10]:
```suggestion
#define CATCH_STOP(expr) OUTCOME_TRY(::fc::adt::catchStop((expr)))
```

File: core/vm/actor/builtin/types/miner/sectors.cpp (hunk 1 above)
Comment (turuslan) [reply core/vm/actor/builtin/types/miner/sectors.cpp:71]: > `// Lotus gas conformance`

it **must** load to charge gas

File: core/adt/stop.hpp (hunk 2 above)
Comment (wer1st) [reply core/adt/stop.hpp:10]: we decided to leave it as is

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: test/core/markets/retrieval/retrieval_market_test.cpp [hunk 3]
```diff
@@ -94,6 +95,9 @@ namespace fc::markets::retrieval::test {
     RetrievalPeer peer{.address = miner_worker_address,
                        .peer_id = host->getId(),
                        .piece = boost::none};
+
+    // TODO(turuslan): fix LSAN
```
Comment (wer1st) [inline test/core/markets/retrieval/retrieval_market_test.cpp:99]: @turuslan please notice this

**Pull request merged**